presto-spi/src/main/java/com/facebook/presto/spi/PrestoException.java

```diff
@@ -45,12 +45,16 @@ public ErrorCode getErrorCode()
     public String getMessage()
     {
         String message = super.getMessage();
+
         if (message == null && getCause() != null) {
             message = getCause().getMessage();
         }
         if (message == null) {
             message = errorCode.getName();
         }
+        if (message.equals("/ by zero")) {
```

Comment:
This is an unusual place to patch this. Where is the division by zero and the arithmetic exception? That's where I'd expect the message to be set.

Reply:
Seems like all the other division by zero errors are handled in my change. The only other place where the code would be changed is arithmeticexception class (which is read only) or in abstract classes which wouldn't work

Reply:
Where do we call the Guava method that initially creates this message and throws this exception? That's where I'd look.

Reply:
@elharo After further research, seems like the best quick fix is to patch it here and then make the PR in guava and wait for them to change it. All the error messages are coming from read only files